gateway/src/error.rs

```diff
@@ -63,6 +63,17 @@ pub enum GatewayError {
         source: io::Error,
     },
 
+    #[error(
+        "failed to load config file for wireguard (gateway-id: '{id}') using path '{}'. detailed message: {source}",
+        path.display()
+    )]
+    WireguardConfigLoadFailure {
+        id: String,
+        path: PathBuf,
+        #[source]
```

Comment:
if the field is named `source` then you don't need the attribute

Reply:
Good point. Then I should probably remove the attribute for the other error types as well.